provisioner/windows-restart/provisioner.go

```diff
@@ -174,28 +174,40 @@ WaitLoop:
 }
 
 var waitForCommunicator = func(p *Provisioner) error {
+	runCustomRestartCheck := true
 	for {
-		cmd := &packer.RemoteCmd{Command: p.config.RestartCheckCommand}
-		var buf, buf2 bytes.Buffer
-		cmd.Stdout = &buf
-		cmd.Stdout = io.MultiWriter(cmd.Stdout, &buf2)
 		select {
 		case <-p.cancel:
 			log.Println("Communicator wait canceled, exiting loop")
 			return fmt.Errorf("Communicator wait canceled")
 		case <-time.After(retryableSleep):
 		}
+		if runCustomRestartCheck == true {
```

Comment:
I think this whole block down to line 194 should be outside of the for loop

Reply:
what I mean is that we don't have to check `if p.config.RestartCheckCommand == DefaultRestartCheckCommand {` inside a loop and we don't need to create `cmdRestartCheck := &packer.RemoteCmd{Command: p.config.RestartCheckCommand}` inside a loop

we still need to run `StartWithUi` inside a loop

Reply:
oh, I see.  duh.  You're right, of course.